osu.Game/Online/Multiplayer/MultiplayerClient.cs

```diff
@@ -420,6 +422,46 @@ Task IMultiplayerClient.UserStateChanged(int userId, MultiplayerUserState state)
             return Task.CompletedTask;
         }
 
+        Task IMultiplayerClient.MatchRulesetUserStateChanged(int userId, MatchRulesetUserState state)
+        {
+            if (Room == null)
+                return Task.CompletedTask;
+
+            Scheduler.Add(() =>
+            {
+                if (Room == null)
+                    return;
+
+                Room.Users.Single(u => u.UserID == userId).MatchRulesetState = state;
+                RoomUpdated?.Invoke();
+            }, false);
+
+            return Task.CompletedTask;
+        }
+
+        Task IMultiplayerClient.MatchRulesetRoomStateChanged(MatchRulesetRoomState state)
+        {
+            if (Room == null)
+                return Task.CompletedTask;
+
+            Scheduler.Add(() =>
+            {
+                if (Room == null)
+                    return;
+
+                Room.MatchRulesetState = state;
+                RoomUpdated?.Invoke();
+            });
```

Comment:
This doesn't specify `forceScheduled = false` like `MatchRulesetUserStateChanged` does above - is that intentional or an oversight?

Reply:
Seems like an oversight. For now I'm just trying to imitate the existing way things are done.